PART_1_SHARE_INFO.md

```diff
@@ -411,17 +423,63 @@ work up to the GitHub repository:
 git push
 ```
 
+#### Personal Access Tokens
+
 :warning: You'll need to authenticate with GitHub at this point. So far all
 you've been doing with things like `git clone` and `git pull` is "reading" info
 from Github, and anyone can do that on a public repository (which this is).
 Now, however, you want to _change_ something, so you need to prove that you
-have permission to make changes to the contents of the repository. How exactly
-this will play out depends on your setup and what exactly GitHub is doing at
-the moment. Hopefully it'll be fairly "obvious" how to authenticate after you
-enter `git push`, but if not definitely ask! If it asks for a username and
-password at some point, that will almost certainly need to be the info you
-used when setting up your GitHub account.
-
+have permission to make changes to the contents of the repository.
+
+To make `git push` work on the command line you'll need [a GitHub _Personal
+Access Token_ (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+GitHub doesn't want you typing your GitHub password on the command line
+(where there's no guarantee that it'll be encrypted), so they instead
+have you get a PAT from their web site, and type that in on the command line
+instead. You don't need PATs on things like GitKraken and VS Code because
+they completely control the communication with GitHub's servers, and GitHub
+can force them to use encrypted communication channels.
+
+PATs are basically short-lived "passwords" that you can use to authenticate with
+GitHub up until they expire. They also have neat properties like the ability
+to constrain a PAT to only work on a particular repository/project, or only
+have certain permissions on that project, all helping limit the damage if
+someone was able to get a copy of your PAT.
+
+To create a PAT for this lab, follow GitHub's ["Creating a fine grained personal access token"](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token)
+instructions. We're going to create a PAT that's entirely specific to this lab,
+which means you'll need to create new PATs later in the semester if/when you
+need them. You can go with the defaults for most of the options, but there will
+be some questions/options you'll need to address when creating your PAT:
+
+- You can use whatever **Token name** makes sense to you for the token.
+- I'd go with an **Expiration** of 7 days since we should be done with this lab
+  the end of the week.
+- Under "Resource owner" you _have_ to select the organization for this semester's
+  Software Design course. That's probably something of the form `UMM-CSci-3601-S24`.
+- Under **Reposity access** choose "Only select repositories". Then choose
```

Comment:
typo - should be Repository rather than Reposity

Reply:
And it even had a blue squiggly underline in VS Code. Doh!